src/index.js

```diff
@@ -0,0 +1,87 @@
+import { InstanceBase, runEntrypoint } from '@companion-module/base'
+import { initActions } from './actions.js'
+import { API } from './api.js'
+import { initFeedbacks } from './feedbacks.js'
+import { initProduct } from './models.js'
+import { setPolling } from './polling.js'
+import { initPresets } from './presets.js'
+import UpgradeScripts from './upgrades.js'
+import { initVariables, checkVariables } from './vars.js'
+import { getConfigDefinitions } from './config.js'
+
+class PTZControlerInstance extends InstanceBase {
+	constructor(internal) {
+		super(internal)
+		this.data = {}
+		this.polling = {}
+		this.product
+	}
+
+	async init(config) {
+		this.data = {
+			camera: 'NaN',
+			group: 'NaN',
+			port: 'NaN',
+		}
+
+		this.config = config
+		this.config.model = this.config.model || 'AW-RP50'
+		this.config.enablePolling = this.config.enablePolling || true
+
+		this.updateStatus('connecting')
+		this.api = new API(config)
+		this.configUpdated(config)
+	}
+
+	async destroy() {
+		if (this.socket !== undefined) {
+			this.socket.destroy()
+		}
+
+		if (this.polling.interval) {
+			clearInterval(this.polling.interval)
+		}
+
+		this.debug('destroy', this.id)
+	}
+
+	async configUpdated(config) {
+		const self = this
+
+		self.config = config
+		this.updateStatus('connecting')
```

Comment:
If this would be a connection based protocol, this line had to go in the if after line 55.
So as this is not connection based, you can either set it to ok or unknown here because you just don't know the status at this point.
Or the better solution wound be to put it with connecting in the if and then send some test command within `api.updateConfig()` to get the valid status right after a host change.

Reply:
I've refactored the status update system. Can you review the updated code?